app/main.py

```diff
@@ -1,2 +1,97 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list[None | tuple[Hashable, hash, Any]] = [None] * 8
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        def add_item_to_table(
```

Comment:
instead of writing a function inside a magic method, make it just a class method